src/dirtyfields/compare.py

```diff
@@ -0,0 +1,36 @@
+import datetime
+import pytz
+import warnings
+
+from django.conf import settings
+from django.utils import timezone
+
+
+def raw_compare(cls, new_value, old_value):
+    return new_value != old_value
+
+
+def time_zone_support_compare(cls, new_value, old_value):
```

Comment:
This function could be a more general purpose one that makes a best effort comparison regardless of the timezone awareness of the input parameters. Currently the logic of the function is obviously adapted to a specific piece of code that calls this and the awareness of the incoming parameters are assumed based on the value of `USE_TZ`.